lib/std/heap.zig

```diff
@@ -8,6 +8,273 @@ const c = std.c;
 const Allocator = std.mem.Allocator;
 const windows = std.os.windows;
 
+const missing_min_page_size = @compileError("The Zig standard library is missing a min_page_size for " ++ @tagName(builtin.cpu.arch) ++ "-" ++ @tagName(builtin.os.tag));
+const missing_max_page_size = @compileError("The Zig standard library is missing a max_page_size for " ++ @tagName(builtin.cpu.arch) ++ "-" ++ @tagName(builtin.os.tag));
+
+/// Compile time minimum page size that the architecture/OS combination supports. Pointers aligned to the system's page size are aligned to at least `min_page_size`, but system calls such as `mmap` and `VirtualAlloc` may return pointers with much larger alignments.
+pub const min_page_size: usize = switch (builtin.os.tag) {
+    .driverkit, .ios, .macos, .tvos, .visionos, .watchos => switch (builtin.cpu.arch) {
+        .x86, .x86_64 => 4 << 10,
+        .thumb, .thumbeb, .arm, .armeb, .aarch64, .aarch64_be => 16 << 10,
+        else => missing_min_page_size,
+    },
+    // -- <https://devblogs.microsoft.com/oldnewthing/20210510-00/?p=105200>
+    .windows => switch (builtin.cpu.arch) {
+        .x86, .x86_64 => 4 << 10,
+        // SuperH => 4 << 10,
+        .mips, .mipsel, .mips64, .mips64el => 4 << 10,
+        .powerpc, .powerpcle, .powerpc64, .powerpc64le => 4 << 10,
+        // DEC Alpha => 8 << 10,
+        // Itanium => 8 << 10,
+        .thumb, .thumbeb, .arm, .armeb, .aarch64, .aarch64_be => 4 << 10,
+        else => missing_min_page_size,
+    },
+    .wasi => switch (builtin.cpu.arch) {
+        .wasm32, .wasm64 => 64 << 10,
+        else => missing_min_page_size,
+    },
+    // https://github.com/tianocore/edk2/blob/b158dad150bf02879668f72ce306445250838201/MdePkg/Include/Uefi/UefiBaseType.h#L180-L187
+    .uefi => 4 << 10,
+    .freestanding => switch (builtin.cpu.arch) {
```

Comment:
I don't think handling `freestanding` in these makes much sense in practice. Freestanding explicitly means that there's no OS, and like I said before, we can only define these constants knowing *both* architecture and OS.

People who are using the standard library in a kernel should compile with the `other` OS tag. Then, if necessary, we can add a bring-your-own-OS hook for the min/max page sizes if that is useful. But I wouldn't worry about that in this PR; we tend to do that on an as-needed basis.